homeassistant/components/intent/timers.py

```diff
@@ -0,0 +1,772 @@
+        if not matching_timers:
+            # No matches
+            return matching_timers
+
+    if "device_id" not in slots:
+        # Can't re-order based on area/floor
+        return matching_timers
+
+    # Use device id to order remaining timers
+    device_id: str = slots["device_id"]["value"]
+    device = timer_manager.device_registry.async_get(device_id)
+    if (device is None) or (device.area_id is None):
+        return matching_timers
+
+    area = timer_manager.area_registry.async_get_area(device.area_id)
+    if area is None:
+        return matching_timers
+
+    def area_floor_sort(timer: TimerInfo) -> int:
+        if timer.area_id == area.id:
+            return -2
+
+        if timer.floor_id == area.floor_id:
+            return -1
+
+        return 0
+
+    matching_timers.sort(key=area_floor_sort)
+
+    return matching_timers
+
+
+def _get_total_seconds(slots: dict[str, Any]) -> int:
+    """Return the total number of seconds from hours/minutes/seconds slots."""
+    total_seconds = 0
+    if "hours" in slots:
+        total_seconds += 60 * 60 * int(slots["hours"]["value"])
+
+    if "minutes" in slots:
+        total_seconds += 60 * int(slots["minutes"]["value"])
+
+    if "seconds" in slots:
+        total_seconds += int(slots["seconds"]["value"])
+
+    return total_seconds
+
+
+class SetTimerIntentHandler(intent.IntentHandler):
+    """Intent handler for starting a new timer."""
+
+    intent_type = intent.INTENT_SET_TIMER
+    slot_schema = {
+        vol.Required(vol.Any("hours", "minutes", "seconds")): cv.positive_int,
+        vol.Optional("name"): cv.string,
+        vol.Optional("device_id"): cv.string,
+        vol.Optional("data"): dict,
+    }
+
+    async def async_handle(self, intent_obj: intent.Intent) -> intent.IntentResponse:
+        """Handle the intent."""
+        hass = intent_obj.hass
+        timer_manager: TimerManager = hass.data[TIMER_DATA]
+        slots = self.async_validate_slots(intent_obj.slots)
+
+        device_id: str | None = None
+        if "device_id" in slots:
+            device_id = slots["device_id"]["value"]
+
+        name: str | None = None
+        if "name" in slots:
+            name = slots["name"]["value"]
+
+        hours: int | None = None
+        if "hours" in slots:
+            hours = int(slots["hours"]["value"])
+
+        minutes: int | None = None
+        if "minutes" in slots:
+            minutes = int(slots["minutes"]["value"])
+
+        seconds: int | None = None
+        if "seconds" in slots:
+            seconds = int(slots["seconds"]["value"])
+
+        data: dict[str, Any] | None = None
+        if "data" in slots:
+            data = slots["data"]["value"]
+
+        await timer_manager.start_timer(
+            hours,
+            minutes,
+            seconds,
+            language=intent_obj.language,
+            device_id=device_id,
+            name=name,
+            data=data,
+        )
+
+        return intent_obj.create_response()
```

Comment:
I wonder if there should be some timer ID attached to the response as metadata, so the UI showing the response can show the timer right away.

Reply:
I'll save this for a follow-up PR when we think more about the frontend.